Ruby-Assignment-8-TDD/roman-number-coding/lib/roman_number.rb

```diff
@@ -0,0 +1,33 @@
+
+class RomanNumber
+  @@mappings = { M: 1000, D: 500, C: 100, L: 50, X:10, V: 5, I: 1 }
+  attr_reader :number_str
+  alias_method :as_string, :number_str
+
+  def initialize(number_str)
+    @number_str = number_str
+  end
+
+  def to_decimal
+    converted_num = 0
+    last_decimal = 0
+    @number_str.split('').reverse.each do |element|
```

Comment:
Try: this loop should reside in another method, and to decimal would give only decimal value for given input.